Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -4848,6 +4873,37 @@ - (void)collectionView:(UICollectionView *)collectionView
     conversationViewCell.isCellVisible = NO;
 }
 
+#pragma mark - ContactsPickerDelegate
+
+- (void)contactsPickerDidCancel:(ContactsPicker *)contactsPicker
+{
+    DDLogDebug(@"%@ in %s", self.logTag, __PRETTY_FUNCTION__);
+}
+
+- (void)contactsPicker:(ContactsPicker *)contactsPicker contactFetchDidFail:(NSError *)error
+{
+    DDLogDebug(@"%@ in %s with error %@", self.logTag, __PRETTY_FUNCTION__, error);
+}
+
+- (void)contactsPicker:(ContactsPicker *)contactsPicker didSelectContact:(Contact *)contact
+{
+    DDLogDebug(@"%@ in %s with contact: %@", self.logTag, __PRETTY_FUNCTION__, contact);
+
+    // TODO actually build contact message.
+    self.inputToolbar.messageText = contact.fullName;
+}
+
+- (void)contactsPicker:(ContactsPicker *)contactsPicker didSelectMultipleContacts:(NSArray<Contact *> *)contacts
+{
+    DDLogDebug(@"%@ in %s with contacts: %@", self.logTag, __PRETTY_FUNCTION__, contacts);
```

Comment:
Shouldn't this be `OWSFail()`?  And shouldn't we configure the contact picker to be "single only" for contact shares?

Reply:
> Shouldn't this be OWSFail()

Yup, added.

> And shouldn't we configure the contact picker to be "single only" for contact shares?

Yup, it is configured via the initializer with the `multiSelection:NO` parameter.